test/test_distributions.py

```diff
@@ -1542,13 +1577,27 @@ def setUp(self):
 
     def test_kl_monte_carlo(self):
         set_rng_seed(0)  # see Note [Randomized statistical tests]
-        for p, q in self.finite_examples:
-            x = p.sample(sample_shape=(23000,))
+        for (p, _), (_, q) in self.finite_examples:
+            x = p.sample(sample_shape=(1000,))
             expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
             actual = kl_divergence(p, q)
-            message = 'Incorrect KL({}, {}). expected {}, actual {}'.format(
-                type(p).__name__, type(q).__name__, expected, actual)
-            self.assertEqual(expected, actual, prec=0.1, message=message)
+            message = 'Incorrect KL({}, {}).\nExpected (Monte Carlo): {}\nActual (analytic): {}'.format(
+                    type(p).__name__, type(q).__name__, expected, actual)
+            try:
```

Comment:
nit: I would remove this try and instead simply test:
```py
if torch.abs(actual - expected).max() > self.precision:
    for sample_extra in self.sampling_extra:
        ...
self.assertEqual(...)
```
In general it's dangerous to `try: ... except` without specifying the exception type being caught. You could e.g. catch a `KeyboardInterrupt` here instead of an `AssertionError`.